File: day2/puzzle1.py
#!/usr/bin/env python3

import re

data_file_name = "puzzle2_data.txt"

with open(data_file_name, 'r') as file:
    directions = [line for line in file]

down_count = 0
up_count = 0
forwards_count = 0

for direction in directions:
    magnitude = int(re.search(r'\d+', direction).group())

    if "down" in direction:
        down_count += magnitude
    elif "up" in direction:
        up_count += magnitude
    elif "forward" in direction:
        forwards_count += magnitude

print(f"Up: {up_count}, down: {down_count}, forward: {forwards_count} = {forwards_count}*({down_count}-{up_count}) = {forwards_count * (down_count-up_count)}")

File: day2/puzzle1.swift
#!/usr/bin/swift

import Foundation

let dataFilePath = "day2_data.txt"

// Read file
let fileContents = try String(contentsOfFile: dataFilePath, encoding: String.Encoding.utf8)
let directions = fileContents.split(whereSeparator: \.isNewline).map( { $0 } ) // Force unwrapping not ideal

// Reduce directions to counts per direction
let downCount = directions.reduce(0) { $0 + ($1.contains("down") ? Int($1.filter("0123456789".contains))! : 0 ) }
let upCount = directions.reduce(0) { $0 + ($1.contains("up") ? Int($1.filter("0123456789".contains))! : 0 ) }
let forwardCount = directions.reduce(0) { $0 + ($1.contains("forward") ? Int($1.filter("0123456789".contains))! : 0) }

print("Up: \(upCount), down: \(downCount), forward: \(forwardCount) = \(forwardCount)*(\(downCount)-\(upCount)) = \(forwardCount * (downCount-upCount))")